### ISSUE TITLE
About validation in pytorch

### ISSUE THREAD

**Issue by soon-will** [thuml/HashNet#22]
Thank you for your effort.

In pytorch version code, the model was saved after 3,000 iterations, so after 10,000 iterations(train) we saved about 3 models. So, I have two questions:

1. Which model are you going to use to test?

2. It seems that you don't val the model after training, so how do you ensure the model saved will be the best model?

Thanks for your reply.

**Comment by caozhangjie** [comment]
1.You can use the model at 6000-7000 iterations.

2.The parameters are tuned in our experiments with cross-validation and I only set them as the optimal parameters for reproducing.

**Comment by soon-will** [comment]
@caozhangjie 
Thanks for your reply.

About the first question, if there is no val set, then how do you ensure the model is the best model? The parameters, like learning rate, can be tuned  with cross-validation, should we use val set to choose the better model , like training a classification model?

**Comment by caozhangjie** [comment]
Yes, you can divide the whole training set into 10 pieces and use 10-fold cross-validation as in classification.

**Comment by soon-will** [comment]
Thank you for your reply.
